pkg/endpointmanager/manager.go

```diff
@@ -405,12 +405,20 @@ func AddEndpoint(owner endpoint.Owner, ep *endpoint.Endpoint, reason string) (er
 	}
 
 	if build {
-		go func() {
+		if asynchronouslyBuild {
```

Comment:
So it seems that for the docker plugin, we have always been waiting for the initial build to complete. It would be better then for the docker plugin to also pass in `SyncBuildEndpoint`. The docker plugin version is tied to the agent version 1:1 so as we change the behavior of `createEndpoint()`, it is fine to require changing the docker plugin as well. In particular as an API user would expect an async build if `SyncBuildEndpoint` is *not* passed in.